tests/unit/lib/s3-nodetest.js

```diff
@@ -284,6 +284,48 @@ describe('s3', function() {
           assert.equal(headParams.Key, expectedKey);
         });
     });
+
+    it('correctly pages s3#listObjects calls when necessary', function() {
+
+      var revisions = [
+        { Key: 'test.html:111', LastModified: new Date('September 27, 2015 01:00:00'), ETag: '111' },
+        { Key: 'test.html:222', LastModified: new Date('September 27, 2015 02:00:00') , ETag: '222' },
+        { Key: 'test.html:333', LastModified: new Date('September 27, 2015 03:00:00') , ETag: '333' },
+        { Key: 'test.html:444', LastModified: new Date('September 27, 2015 04:00:00') , ETag: '444' },
+        { Key: 'test.html:555', LastModified: new Date('September 27, 2015 05:00:00') , ETag: '555' },
+        { Key: 'test.html:666', LastModified: new Date('September 27, 2015 06:00:00') , ETag: '666' },
+        { Key: 'test.html:777', LastModified: new Date('September 27, 2015 07:00:00') , ETag: '777' },
+        { Key: 'test.html:888', LastModified: new Date('September 27, 2015 08:00:00') , ETag: '888' },
+        { Key: 'test.html:999', LastModified: new Date('September 27, 2015 09:00:00') , ETag: '999' },
+        { Key: 'test.html:000', LastModified: new Date('September 27, 2015 10:00:00') , ETag: '000' },
+      ];
+
+      var resultsPerPage = 4;
+      var numCalls = 0;
+
+      s3Client.listObjects = function(params, cb) {
```

Comment:
can we please update this stub to mirror the S3 response as close as possible? For a truncated responses we should also return the `NextMarker` property for example

it would also be good to assert on the correct params passed do this function (i.e. assert that we are including the `Marker` option as expected)